### PR TITLE
FieldSpec

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
@Emilgardis What do you think about this?

File: src/generate/generic.rs [hunk 1]
```diff
@@ -47,6 +50,12 @@ pub trait RegisterSpec {
     type Ux: RawReg;
 }
 
+/// Raw field type
+pub trait FieldSpec: Sized {
+    /// Raw field type (`u8`, `u16`, `u32`, ...).
+    type Ux: Copy + PartialEq + From<Self>;
+}
+
```
Comment (Emilgardis) [inline src/generate/generic.rs:58]: why not make this a supertrait?

```suggestion
/// Raw field type
pub trait FieldSpec: Copy + PartialEq + From<Self> + Sized {}

```

Comment (burrbull) [reply src/generate/generic.rs:58]: `Self` and `Ux` are different types if field has enumeratedValues.

**Comment by burrbull** [comment]
@Emilgardis So what your judgment?

**Review by Emilgardis — DISMISSED** [review]
This is good, no concerns from me except maybe the name choice of `Ux`, but we're already using that so it's fine.

feel free to r=me

**Comment by burrbull** [comment]
bors r=@Emilgardis

**Comment by burrbull** [comment]
bors retry

**Comment by burrbull** [comment]
bors r=@Emilgardis

**Comment by Emilgardis** [comment]
bors retry

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**